DancingGoat/Helpers/SampleDeliveryClient.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DancingGoat.Controllers;
+using DancingGoat.Models;
+using DancingGoat.Utils;
+using KenticoCloud.Delivery;
+using KenticoCloud.Delivery.InlineContentItems;
+using Newtonsoft.Json.Linq;
+
+namespace DancingGoat.Helpers
+{
+    public class SampleDeliveryClient : IDeliveryClient
+    {
+        private DeliveryClient _deliveryClient;
+        private readonly List<SampleSiteFilter> _sampleSiteFilter = new List<SampleSiteFilter> { new SampleSiteFilter() };
+
+        public SampleDeliveryClient()
+        {
+            // Use the provider to get environment variables.
+            var provider = new ConfigurationManagerProvider();
+
+            // Build DeliveryOptions with default or explicit values.
+            var options = provider.GetDeliveryOptions();
+
+            options.ProjectId = ProjectUtils.GetProjectId();
+
+            _deliveryClient = new DeliveryClient(options);
+            _deliveryClient.CodeFirstModelProvider.TypeProvider = new CustomTypeProvider();
+            _deliveryClient.ContentLinkUrlResolver = new CustomContentLinkUrlResolver();
+        }
+
+        public Task<JObject> GetItemJsonAsync(string codename, params string[] parameters)
+        {
+            parameters.ToList().Add(_sampleSiteFilter.FirstOrDefault().GetQueryStringParameter());

```

Comment:
parameters could be null so this might end with null reference exception, you should start with _sampleSiteFilter as LanguageClient does in all below cases